core/src/main/scala/org/apache/spark/SparkContext.scala

```diff
@@ -370,6 +370,7 @@ class SparkContext(config: SparkConf) extends Logging with ExecutorAllocationCli
   taskScheduler.start()
 
   val applicationId: String = taskScheduler.applicationId()
+  val applicationAttemptId: String = taskScheduler.applicationAttemptId()
```

Comment:
Just kept it on similar lines as applicationId. An application with different attempt id, is similar to different application in many ways.